### ISSUE TITLE
[Notifications 1.0.0] BUILD FAILURE: Build failed with an exception. Execution failed for task ':app:mergeDebugAssets'. This project uses AndroidX dependencies, but the 'android.useAndroidX' property is not enabled. Set this property to true in the gradle.properties file and retry.

### ISSUE THREAD

**Issue by UsabilitySpace** [cph-cachet/flutter-plugins#180]
### EMULATOR AND OS: 

 - **Emulator**: `Android Studio Emulator` version `Nexus 5X API 28` = `qemu-System-x86_64`
 - **OS**: `iOs` version `10.15.6`

### ISSUE DESCRIPTION: 

After downloading, and when trying to build `flutter-plugins/packages/notifications` project, such a build failure occurs:

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:mergeDebugAssets'.
> This project uses AndroidX dependencies, but the 'android.useAndroidX' property is not enabled. Set this property to true in the gradle.properties file and retry.
  The following AndroidX dependencies are detected: androidx.core:core:1.1.0, androidx.fragment:fragment:1.1.0, androidx.versionedparcelable:versionedparcelable:1.1.0, androidx.lifecycle:lifecycle-common:2.2.0, androidx.lifecycle:lifecycle-common-java8:2.2.0, androidx.customview:customview:1.0.0, androidx.lifecycle:lifecycle-livedata:2.0.0, androidx.lifecycle:lifecycle-viewmodel:2.1.0, androidx.loader:loader:1.0.0, androidx.lifecycle:lifecycle-livedata-core:2.0.0, androidx.activity:activity:1.0.0, androidx.collection:collection:1.1.0, androidx.viewpager:viewpager:1.0.0, androidx.arch.core:core-runtime:2.0.0, androidx.arch.core:core-common:2.1.0, androidx.savedstate:savedstate:1.0.0, androidx.annotation:annotation:1.1.0, androidx.lifecycle:lifecycle-runtime:2.2.0

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3m 57s
AndroidX incompatibilities may have caused this build to fail. Please migrate your app to AndroidX. See https://goo.gl/CP92wY .
Exception: Gradle task assembleDebug failed with exit code 1
``` 

### STEPS TO REPRODUCE: 

1. In Android Studio, choose `File > Open` to open `flutter-plugins/packages/notifications` directory as an existing project.

2. Chose `Run > Edit Configurations...` and then add new Flutter configuration that targets to run `example/lib/main.dart`.

3. According to [https://goo.gl/CP92wY](https://goo.gl/CP92wY) ⇒ [https://flutter.dev/docs/development/androidx-migration](https://flutter.dev/docs/development/androidx-migration), migrate to AndroidX those two android subprojects: 

- `flutter-plugins/packages/notifications/example/android`
- `flutter-plugins/packages/notifications/android`

4. Run `Nexus 5X API 28` = `qemu-System-x86_64` Android Studio Emulator.

5. To start building project, press `⌃R` = `Ctrl+R` key combination, or press `Run` button in the Android Studio toolbar: 

![image](https://user-images.githubusercontent.com/46682521/91633266-17a11e00-ea11-11ea-94fb-eed0b6978b2a.png)

### OTHER STEPS TO REPRODUCE: 

The issue persists even if trying the next slightly different steps 1'-4' to reproduce:

1'. In Android Studio, choose `File > Open` to open `flutter-plugins/packages/notifications/example` directory as an existing project (but not `flutter-plugins/packages/notifications/` directory).

2'. = Step 3 (see above). 

3'. = Step 4 (see above). 

4'. = Step 5 (see above). 

![image](https://user-images.githubusercontent.com/46682521/91633266-17a11e00-ea11-11ea-94fb-eed0b6978b2a.png)

### EXPECTED BEHAVIOUR:

Project should build & run successfully. 

### ACTUAL BEHAVIOUR: 

Project doesn't even build. 

### SCREENSHOTS:

![image](https://user-images.githubusercontent.com/46682521/91633353-cba2a900-ea11-11ea-8a59-ca65a2c6b6f4.png)

### FLUTTER DOCTOR:

```
flutter-plugins/packages/notifications > flutter doctor -v

[✓] Flutter (Channel master, 1.22.0-2.0.pre.36, on Mac OS X 10.15.6 19G73, locale ru-RU)
    • Flutter version 1.22.0-2.0.pre.36 at /Applications/flutter
    • Framework revision d30e36ba8c (7 дней назад), 2020-08-21 22:36:03 -0400
    • Engine revision d495da20d0
    • Dart version 2.10.0 (build 2.10.0-49.0.dev)

 [✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/***********/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.6, Build version 11E708
    • CocoaPods version 1.8.4

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 48.1.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
 
[✓] Connected device (2 available)            
    • AOSP on IA Emulator (mobile) • emulator-5554 • android-x86 • Android 9 (API 28) (emulator)
    • macOS (desktop)              • macos         • darwin-x64  • Mac OS X 10.15.6 19G73

• No issues found!
```

### NOTE: 

- Running flutter clean doesn't help to overcome the issue.

### THE AIM OF HANDLING THIS ISSUE: 

What are reproducable and exact steps for Android Studio users to overcome this issue in Android Studio under iOS?

**Comment by UsabilitySpace** [comment]
### HOW TO HADLE THIE ISSUE: 

1. Open `flutter-plugins/packages/notifications/example/android/gradle.properties` file. 

2. Replace these two lines: 

      `org.gradle.jvmargs=-Xmx1536M`
      `android.enableR8=true`

    ... with these four lines: 

      `org.gradle.jvmargs=-Xmx1536M`
      `android.enableR8=true`
      `android.useAndroidX=true`
      `android.enableJetifier=true`

### WARNING: 

Overcoming the issue causes new severe build failure: 

```
──────────────────────────────────────────────────────────────────────────────
Your Flutter application is created using an older version of the Android
embedding. It's being deprecated in favor of Android embedding v2. Follow the
steps at

https://flutter.dev/go/android-project-migration

to migrate your project.
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

Launching lib/main.dart on AOSP on IA Emulator in debug mode...
Running Gradle task 'assembleDebug'...

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:processDebugResources'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
   > Android resource linking failed
     /Users/**************/.gradle/caches/transforms-2/files-2.1/cf83f438fd751d9ff271198248ec750b/core-1.1.0/res/values/values.xml:142:5-173:25: AAPT: error: resource android:attr/fontVariationSettings not found.

     /Users/**************/.gradle/caches/transforms-2/files-2.1/cf83f438fd751d9ff271198248ec750b/core-1.1.0/res/values/values.xml:142:5-173:25: AAPT: error: resource android:attr/ttcIndex not found.


* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 58s
The built failed likely due to AndroidX incompatibilities in a plugin. The tool is about to try using Jetfier to solve the incompatibility.
Building plugin notifications...
Running Gradle task 'assembleAarRelease'...


FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':compileReleaseAidl'.
> SDK location not found. Define location with an ANDROID_SDK_ROOT environment variable or by setting the sdk.dir path in your project's local properties file at 'flutter-plugins/packages/notifications/android/local.properties'.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2s

Exception: The plugin notifications could not be built due to the issue above.

```

### Later I'll create a separate issue for that new build failure and provide a link, if needed.

**Comment by UsabilitySpace** [comment]
### HOW TO OVERCOME SEVERE BUILD FAILURE (X):

To partially overcome that severe build failure `(X)` [see above] you should migrate both these your projects to AndroidX: 

- `flutter-plugins/packages/notifications/example/android`
- `flutter-plugins/packages/notifications/android`

### ANOTHER SEVERE BUILD FAILURE (Y):

Though, overcoming severe build failure `(X)` causes one more severe build failure `(Y)`: 

```
(...)
/Applications/flutter/packages/flutter/lib/src/rendering/flow.dart:386:50: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.10 or higher.
        hitTest: (BoxHitTestResult result, Offset? position) {
                                                 ^
/Applications/flutter/packages/flutter/lib/src/rendering/flow.dart:386:44: Error: 'Offset' isn't a type.
        hitTest: (BoxHitTestResult result, Offset? position) {
                                           ^^^^^^
/Applications/flutter/packages/flutter/lib/src/rendering/flow.dart:387:58: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.10 or higher.
          return child.hitTest(result, position: position!);
                                                         ^
/Applications/flutter/packages/flutter/lib/src/rendering/flow.dart:385:19: Error: The getter 'position' isn't defined for the class 'RenderFlow'.
 - 'RenderFlow' is from 'package:flutter/src/rendering/flow.dart' ('/Applications/flutter/packages/flutter/lib/src/rendering/flow.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'position'.
        position: position,
                  ^^^^^^^^
/Applications/flutter/packages/flutter/lib/src/rendering/flow.dart:387:40: Error: No named parameter with the name 'position'.
          return child.hitTest(result, position: position!);
                                       ^^^^^^^^
/Applications/flutter/packages/flutter/lib/src/rendering/flow.dart:384:9: Error: No named parameter with the name 'transform'.
        transform: transform,
        ^^^^^^^^^
/Applications/flutter/packages/flutter/lib/src/rendering/flow.dart:400:52: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.10 or higher.
      transform.multiply(childParentData._transform!);
                                                   ^
/Applications/flutter/packages/flutter/lib/src/rendering/image.dart:144:45: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.10 or higher.
      _colorFilter = ColorFilter.mode(_color!, _colorBlendMode ?? BlendMode.srcIn);
                                            ^
/Applications/flutter/packages/flutter/lib/src/rendering/image.dart:330:13: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.10 or higher.
      _image!.width.toDouble() / _scale,
            ^
/Applications/flutter/packages/flutter/lib/src/rendering/image.dart:331:13: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.10 or higher.
      _image!.height.toDouble() / _scale,
            ^
/Applications/flutter/packages/flutter/lib/src/rendering/image.dart:381:20: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.10 or higher.
      image: _image!,
                   ^
/Applications/flutter/packages/flutter/lib/src/rendering/image.dart:386:36: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.10 or higher.
      alignment: _resolvedAlignment!,
                                   ^
/Applications/flutter/packages/flutter/lib/src/rendering/image.dart:389:42: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.10 or higher.
      flipHorizontally: _flipHorizontally!,
                                         ^
/Applications/flutter/packages/flutter/lib/src/rendering/image.dart:379:7: Error: No named parameter with the name 'canvas'.
      canvas: context.canvas,
      ^^^^^^
/Applications/flutter/packages/flutter/lib/src/painting/decoration_image.dart:410:6: Context: Found this candidate, but the arguments don't match.
void paintImage({
     ^^^^^^^^^^
/Applications/flutter/packages/flutter/lib/src/rendering/image.dart:409:55: Error: No named parameter with the name 'value'.
    properties.add(FlagProperty('matchTextDirection', value: matchTextDirection, ifTrue: 'match text direction'));
                                                      ^^^^^
/Applications/flutter/packages/flutter/lib/src/foundation/diagnostics.dart:2136:3: Context: Found this candidate, but the arguments don't match.
  FlagProperty(
  ^^^^^^^^^^^^
/Applications/flutter/packages/flutter/lib/src/rendering/layer.dart:105:63: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.10 or higher.
  ContainerLayer? get parent => super.parent as ContainerLayer?;
                                                              ^
/Applications/flutter/packages/flutter/lib/src/rendering/layer.dart:170:9: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.10 or higher.
    bool? result;
(...)
```

**And that ↑ feed is very very very very very long.**

**Comment by UsabilitySpace** [comment]
### HOW I TRIED TO HANDLE THIS ISSUE — VARIANT 2:

0. Delete and reload the whole project from GitHub: [https://github.com/cph-cachet/flutter-plugins](https://github.com/cph-cachet/flutter-plugins). 

1. In Android Studio, choose `File > Open` to open `flutter-plugins/packages/notifications/example` directory as an existing project (but not `flutter-plugins/packages/notifications/` directory).

2. **DO NOT** migrate to AndroidX the next two android sub-projects:

    - `flutter-plugins/packages/notifications/example/android`
    - `flutter-plugins/packages/notifications/android`

3. In `flutter-plugins/packages/notifications/example/android/gradle.properties` file, replace these two lines: 

      `org.gradle.jvmargs=-Xmx1536M`
      `android.enableR8=true`

    ... with these four lines: 

      `org.gradle.jvmargs=-Xmx1536M`
      `android.enableR8=true`
      `android.useAndroidX=true`
      `android.enableJetifier=true`

4. Run `Nexus 5X API 28` = `qemu-System-x86_64` Android Studio Emulator.

5. To start building project, press `⌃R` = `Ctrl+R` key combination, or press `Run` button in the Android Studio toolbar: 

### YET ANOTHER SEVERE BUILD FAILURE (Z): 

Though, variant 2 leads to the next severe build failure `(Z)`: 

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:processDebugResources'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
   > Android resource linking failed
     /Users/***********/.gradle/caches/transforms-2/files-2.1/cf83f438fd751d9ff271198248ec750b/core-1.1.0/res/values/values.xml:142:5-173:25: AAPT: error: resource android:attr/fontVariationSettings not found.

     /Users//***********/.gradle/caches/transforms-2/files-2.1/cf83f438fd751d9ff271198248ec750b/core-1.1.0/res/values/values.xml:142:5-173:25: AAPT: error: resource android:attr/ttcIndex not found.


* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 53s
The built failed likely due to AndroidX incompatibilities in a plugin. The tool is about to try using Jetfier to solve the incompatibility.
✏️  Creating `android/settings_aar.gradle`...
✓ `android/settings_aar.gradle` created successfully.
Building plugin notifications...
Running Gradle task 'assembleAarRelease'...


FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':compileReleaseAidl'.
> SDK location not found. Define location with an ANDROID_SDK_ROOT environment variable or by setting the sdk.dir path in your project's local properties file at 'flutter-plugins/packages/notifications/android/local.properties'.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2s

Exception: The plugin notifications could not be built due to the issue above.

```

**Comment by UsabilitySpace** [comment]
### HOW I TRIED TO HANDLE THIS ISSUE — VARIANT 3: 

After trying `Variant 2` (see above), in `flutter-plugins/packages/notifications/example/pubspec.yaml` file change the next 2 lines: 

```
  notifications: 
    path: ../
```

... with the next 2 lines: 

```
  notifications: ^1.0.0
    # path: ../
```

In fact, that ↑ makes your `notifications` plugin to be included into your `sample` app from `pub.dev` repository rather than directly from your local directory. 

### THE PREVIOUS STEP LEADS TO YET EVEN ONE MORE SEVERE BUILD FAILURE: 

```
The plugin `notifications` requires your app to be migrated to the Android embedding v2. Follow the steps on https://flutter.dev/go/android-project-migration and re-run this command.

```

### FOLLOWING THE STEPS AS SUGGESTED:

Following the steps as suggested here [https://flutter.dev/go/android-project-migration](https://flutter.dev/go/android-project-migration) leads to another build error which is no less severe then all previous: 

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:processDebugResources'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
   > Android resource linking failed
     flutter-plugins/packages/notifications/example/android/app/src/main/res/values/styles.xml:9:5-11:13: AAPT: error: resource drawable/[your_normal_background_drawable] (aka cachet.plugins.notificationsexample:drawable/[your_normal_background_drawable]) not found.


* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 59s
The built failed likely due to AndroidX incompatibilities in a plugin. The tool is about to try using Jetfier to solve the incompatibility.
Building plugin notifications...
Running Gradle task 'assembleAarRelease'...


FAILURE: Build failed with an exception.

* What went wrong:
Task 'assembleAarRelease' not found in root project 'notifications'.

* Try:
Run gradlew tasks to get a list of available tasks. Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 33s

Exception: The plugin notifications could not be built due to the issue above.
```

**Comment by UsabilitySpace** [comment]
### HOW TO FINALIZE THIS ISSUE: 

**1. In Android Studio, __create new flutter app__ (and, as always, don't forget to press `pub get`)**.

**2. Integrate your `notifications` plugin into that new flutter app as described here [https://pub.dev/packages/notifications](https://pub.dev/packages/notifications) in sections `Installation` and `Usage`**. 

**3. Try to `Run` ![image](https://user-images.githubusercontent.com/46682521/91633266-17a11e00-ea11-11ea-94fb-eed0b6978b2a.png) the project in order to see this exception**: 

```
Launching lib/main.dart on Android SDK built for x86 in debug mode...
Running Gradle task 'assembleDebug'...
✓ Built build/app/outputs/flutter-apk/app-debug.apk.
Installing build/app/outputs/flutter-apk/app.apk...
Waiting for Android SDK built for x86 to report its views...
Debug service listening on ws://127.0.0.1:50164/fxhbGMCPcuI=/ws
Syncing files to device Android SDK built for x86...
D/EGL_emulation( 8646): eglMakeCurrent: 0xdf2da900: ver 3 0 (tinfo 0xc01717b0)
I/OpenGLRenderer( 8646): Davey! duration=8170ms; Flags=1, IntendedVsync=4564255763161, Vsync=4564255763161, OldestInputEvent=9223372036854775807, NewestInputEvent=0, HandleInputStart=4564258389060, AnimationStart=4564258408392, PerformTraversalsStart=4564258410359, DrawStart=4572191307324, SyncQueued=4572277288321, SyncStart=4572277956984, IssueDrawCommandsStart=4572278154869, SwapBuffers=4572399905416, FrameCompleted=4572427074562, DequeueBufferDuration=21248000, QueueBufferDuration=270000, 
I/Choreographer( 8646): Skipped 489 frames!  The application may be doing too much work on its main thread.
D/EGL_emulation( 8646): eglMakeCurrent: 0xdf2dc040: ver 3 0 (tinfo 0xd720f770)
E/EventChannel#notifications.eventChannel( 8646): Failed to open event stream
E/EventChannel#notifications.eventChannel( 8646): android.util.AndroidRuntimeException: Calling startActivity() from outside of an Activity  context requires the FLAG_ACTIVITY_NEW_TASK flag. Is this really what you want?
E/EventChannel#notifications.eventChannel( 8646): 	at android.app.ContextImpl.startActivity(ContextImpl.java:952)
E/EventChannel#notifications.eventChannel( 8646): 	at android.app.ContextImpl.startActivity(ContextImpl.java:928)
E/EventChannel#notifications.eventChannel( 8646): 	at android.content.ContextWrapper.startActivity(ContextWrapper.java:383)
E/EventChannel#notifications.eventChannel( 8646): 	at dk.cachet.notifications.NotificationsPlugin.onListen(NotificationsPlugin.kt:46)
E/EventChannel#notifications.eventChannel( 8646): 	at io.flutter.plugin.common.EventChannel$IncomingStreamRequestHandler.onListen(EventChannel.java:188)
E/EventChannel#notifications.eventChannel( 8646): 	at io.flutter.plugin.common.EventChannel$IncomingStreamRequestHandler.onMessage(EventChannel.java:167)
E/EventChannel#notifications.eventChannel( 8646): 	at io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart(DartMessenger.java:85)
E/EventChannel#notifications.eventChannel( 8646): 	at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:692)
E/EventChannel#notifications.eventChannel( 8646): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/EventChannel#notifications.eventChannel( 8646): 	at android.os.MessageQueue.next(MessageQueue.java:336)
E/EventChannel#notifications.eventChannel( 8646): 	at android.os.Looper.loop(Looper.java:174)
E/EventChannel#notifications.eventChannel( 8646): 	at android.app.ActivityThread.main(ActivityThread.java:7356)
E/EventChannel#notifications.eventChannel( 8646): 	at java.lang.reflect.Method.invoke(Native Method)
E/EventChannel#notifications.eventChannel( 8646): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
E/EventChannel#notifications.eventChannel( 8646): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)

════════ Exception caught by services library ══════════════════════════════════════════════════════
The following PlatformException was thrown while activating platform stream on channel notifications.eventChannel:
PlatformException(error, Calling startActivity() from outside of an Activity  context requires the FLAG_ACTIVITY_NEW_TASK flag. Is this really what you want?, null, null)

When the exception was thrown, this was the stack: 
#0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:582:7)
#1      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:159:18)
<asynchronous suspension>
#2      MethodChannel.invokeMethod (package:flutter/src/services/platform_channel.dart:332:12)
#3      EventChannel.receiveBroadcastStream.<anonymous closure> (package:flutter/src/services/platform_channel.dart:540:29)
...
════════════════════════════════════════════════════════════════════════════════════════════════════

```

... **and blank app's screen**: 

![image](https://user-images.githubusercontent.com/46682521/91640676-87cb9600-ea49-11ea-930c-e62334d2a69a.png)

**3. Go to `Edit > Find > Find in Path...` or press `⇧⌘F`=`SHIFT+CMD+F` to open `Find in path master`.**

**4. In `Find in path` master, type `startActivity` and go to the next found file:**

`/Applications/flutter/.pub-cache/hosted/pub.dartlang.org/notifications-1.0.0/android/src/main/kotlin/dk/cachet/notifications/NotificationsPlugin.kt`

**5. In `...NotificationsPlugin.kt` file, at about line # 46, replace this line:**

```
 context!!.startActivity(Intent(ACTION_NOTIFICATION_LISTENER_SETTINGS))
```

**... with the next 3 lines:**

```
   // from: https://stackoverflow.com/questions/3918517/calling-startactivity-from-outside-of-an-activity-context :
   context!!.startActivity(Intent(ACTION_NOTIFICATION_LISTENER_SETTINGS).addFlags(Intent.FLAG_ACTIVITY_NEW_TASK))
   // context!!.startActivity(Intent(ACTION_NOTIFICATION_LISTENER_SETTINGS))
```

**6. Call `flutter clean` in your projects directory to clear all your build caches.**

**7. Run project ![image](https://user-images.githubusercontent.com/46682521/91633266-17a11e00-ea11-11ea-94fb-eed0b6978b2a.png), and give the app all the required permissions:**

![image](https://user-images.githubusercontent.com/46682521/91641201-ffe78b00-ea4c-11ea-9aea-accec73ad967.png)

![image](https://user-images.githubusercontent.com/46682521/91641203-0413a880-ea4d-11ea-921f-405b44410aa1.png)

![image](https://user-images.githubusercontent.com/46682521/91641151-bac35900-ea4c-11ea-9c38-f4b1d3e3b13f.png)

**8. After that, you'll see this:** 

![image](https://user-images.githubusercontent.com/46682521/91641227-1db4f000-ea4d-11ea-8ae0-749b57b7f3f9.png)

**..., and you can read such a log:**

```
event sent after app closed: {id: 0, progressId: null, message: Running "flutter pub get" in example2...}
event sent after app closed: {id: 0, progressId: null, finished: true}
Launching lib/main.dart on Android SDK built for x86 in debug mode...
Running Gradle task 'assembleDebug'...
✓ Built build/app/outputs/flutter-apk/app-debug.apk.
Installing build/app/outputs/flutter-apk/app.apk...
Waiting for Android SDK built for x86 to report its views...
Debug service listening on ws://127.0.0.1:60563/U0TBQldiFzY=/ws
Syncing files to device Android SDK built for x86...
D/EGL_emulation(11848): eglMakeCurrent: 0xdf36cec0: ver 3 0 (tinfo 0xd720f720)
D/EGL_emulation(11848): eglMakeCurrent: 0xd721a120: ver 3 0 (tinfo 0xd720f6c0)
D/AndroidRuntime(11848): Shutting down VM
E/AndroidRuntime(11848): FATAL EXCEPTION: main
E/AndroidRuntime(11848): Process: com.jvhaile.notifications.notifications_example, PID: 11848
E/AndroidRuntime(11848): java.lang.RuntimeException: Unable to instantiate service cachet.plugins.notifications.NotificationListener: java.lang.ClassNotFoundException: Didn't find class "cachet.plugins.notifications.NotificationListener" on path: DexPathList[[zip file "/data/app/com.jvhaile.notifications.notifications_example-8QpPsluCUFcGqfbhpAiudg==/base.apk"],nativeLibraryDirectories=[/data/app/com.jvhaile.notifications.notifications_example-8QpPsluCUFcGqfbhpAiudg==/lib/x86, /data/app/com.jvhaile.notifications.notifications_example-8QpPsluCUFcGqfbhpAiudg==/base.apk!/lib/x86, /system/lib, /system/product/lib]]
E/AndroidRuntime(11848): 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:3940)
E/AndroidRuntime(11848): 	at android.app.ActivityThread.access$1500(ActivityThread.java:219)
E/AndroidRuntime(11848): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1875)
E/AndroidRuntime(11848): 	at android.os.Handler.dispatchMessage(Handler.java:107)
E/AndroidRuntime(11848): 	at android.os.Looper.loop(Looper.java:214)
E/AndroidRuntime(11848): 	at android.app.ActivityThread.main(ActivityThread.java:7356)
E/AndroidRuntime(11848): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(11848): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
E/AndroidRuntime(11848): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
E/AndroidRuntime(11848): Caused by: java.lang.ClassNotFoundException: Didn't find class "cachet.plugins.notifications.NotificationListener" on path: DexPathList[[zip file "/data/app/com.jvhaile.notifications.notifications_example-8QpPsluCUFcGqfbhpAiudg==/base.apk"],nativeLibraryDirectories=[/data/app/com.jvhaile.notifications.notifications_example-8QpPsluCUFcGqfbhpAiudg==/lib/x86, /data/app/com.jvhaile.notifications.notifications_example-8QpPsluCUFcGqfbhpAiudg==/base.apk!/lib/x86, /system/lib, /system/product/lib]]
E/AndroidRuntime(11848): 	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:196)
E/AndroidRuntime(11848): 	at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
E/AndroidRuntime(11848): 	at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
E/AndroidRuntime(11848): 	at android.app.AppComponentFactory.instantiateService(AppComponentFactory.java:129)
E/AndroidRuntime(11848): 	at androidx.core.app.CoreComponentFactory.instantiateService(CoreComponentFactory.java:66)
E/AndroidRuntime(11848): 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:3935)
E/AndroidRuntime(11848): 	... 8 more
I/Process (11848): Sending signal. PID: 11848 SIG: 9
Lost connection to device.
```

**Comment by thomasnilsson** [comment]
Should be fixed in version 1.0.1 :)
